SQL/0000-00-05-ElectrophysiologyTables.sql

```diff
@@ -201,6 +200,87 @@ CREATE TABLE `physiological_archive` (
     ON DELETE CASCADE
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
+-- SQL tables for BIDS derivative file structure
+-- Create physiological_annotation_file_type table
+CREATE TABLE `physiological_annotation_file_type` (
+    `FileType`        VARCHAR(20)   NOT NULL UNIQUE,
+    `Description` VARCHAR(255),
+    PRIMARY KEY (`FileType`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Create physiological_annotation_file table
+CREATE TABLE `physiological_annotation_file` (
+    `AnnotationFileID`    INT(10)      UNSIGNED NOT NULL AUTO_INCREMENT,
+    `PhysiologicalFileID` INT(10)      UNSIGNED NOT NULL,
+    `FileType`            VARCHAR(20)  NOT NULL,
+    `FilePath`            VARCHAR(255) NOT NULL,
+    PRIMARY KEY (`AnnotationFileID`),
+    CONSTRAINT `FK_phys_file_ID`
+        FOREIGN KEY (`PhysiologicalFileID`)
+        REFERENCES `physiological_file` (`PhysiologicalFileID`),
+    CONSTRAINT `FK_annotation_file_type`
+        FOREIGN KEY (`FileType`)
+        REFERENCES `physiological_annotation_file_type` (`FileType`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Create annotation_archive which will store archives of all the annotation files for
+-- Front-end download
+CREATE TABLE `physiological_annotation_archive` (
+  `AnnotationArchiveID` INT(10) UNSIGNED NOT NULL   AUTO_INCREMENT,
+  `PhysiologicalFileID` INT(10) UNSIGNED NOT NULL,
+  `Blake2bHash`         VARCHAR(128)     NOT NULL,
+  `FilePath`            VARCHAR(255)     NOT NULL,
+  PRIMARY KEY (`AnnotationArchiveID`),
+  CONSTRAINT `FK_physiological_file_ID`
+    FOREIGN KEY (`PhysiologicalFileID`)
+    REFERENCES `physiological_file` (`PhysiologicalFileID`)
+    ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Create annotation_parameter table
+-- Note: This corresponds with the JSON annotation files
+CREATE TABLE `physiological_annotation_parameter` (
+    `AnnotationParameterID` INT(10)      UNSIGNED NOT NULL AUTO_INCREMENT,
+    `AnnotationFileID`      INT(10)      UNSIGNED NOT NULL,
+    `Sources`               VARCHAR(255),
+    `Author`                VARCHAR(50),
+    PRIMARY KEY (`AnnotationParameterID`),
+    CONSTRAINT `FK_annotation_file_ID`
+        FOREIGN KEY (`AnnotationFileID`)
+        REFERENCES `physiological_annotation_file` (`AnnotationFileID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Create an annotation_label_type table
+CREATE TABLE `physiological_annotation_label` (
+  `AnnotationLabelID`    INT(5)       UNSIGNED NOT NULL      AUTO_INCREMENT,
+  `LabelName`            VARCHAR(255)          NOT NULL      UNIQUE,
```

Comment:
Why not making `LabelName` the PK?

Reply:
i think it's fine, it's possible the label name spellings might change over time.

Reply:
I agree with Christine, this makes it more flexible to change the names as they get edited.

Reply:
I understand. I made this suggestion after looking at a few examples from the specs, where `LabelName` seems to be used as a key and `LabelDescription` the human-readable version.
```
  "LabelDescription":	{
    “eyes_closed”: “subject has the eyes closed”,
    “eyes_open”: “subject has the eyes open”,
    “watching_tv”: “subject is watching tv”
}
```

Reply:
Yes that is the format of the `.json` file, but I thought that for the database tables it would make more sense to use an ID.